pkg/database/reference.go

```diff
@@ -126,32 +124,35 @@ func (d *db) VerifiableSetReference(req *schema.VerifiableReferenceRequest) (*sc
 		return nil, store.ErrIllegalArguments
 	}
 
-	txMetatadata, err := d.SetReference(req.ReferenceRequest)
+	// Preallocate tx buffers
+	lastTx, err := d.allocTx()
```

Comment:
it may be moved some lines below, right before its use

Reply:
Actually this was moved up on purpose. I wanted to minimize the risk of a scenario where a successful write is made but the whole operation fails because the read tx holder pool was exhausted. In such case the user would get the max concurrency limit exceeded error and could retry the write in case of such error.